crates/autopilot/src/domain/fee/policy.rs

```diff
@@ -0,0 +1,142 @@
+//! Protocol fee implementation.
+//!
+//! The protocol fee is a fee that is defined by the protocol and for each order
+//! we define the way to calculate the protocol fee based on the configuration
+//! parameters.
+
+use {
+    crate::{
+        boundary::{self, Order, OrderClass, OrderUid},
+        infra::{self},
+    },
+    std::{
+        collections::HashMap,
+        sync::{Arc, RwLock},
+    },
+};
+
+/// Protocol fee policies with cache being updated on each auction.
+#[derive(Debug)]
+pub struct PoliciesCache {
```

Comment:
I'm not a fan of adding this policies cache. A cache is usually used for time sensitive operations. In this case loading the quotes from the database may be the bottleneck, so I think if we wanted to build a cache it should be for the quote fetching operation (such a cache may be useful for other places). Then this component would use a cache instead of the database.

Letting the autopilot operate on a policy cache adds unnecessary conceptual complexity imo.

Fee policies just need to be applied or derived from a given order and I think this is exactly the interface the fee domain component should offer.

`fn(order) -> [FeePolicy]`

or event better

`fn(order) -> order` (return order with fee policies)

Reply:
Removed the caching feature. Тhe current interface is 
fn(orders) -> HashMap(order, policies) to reuse the same db connection for thousands of calls.

Reply:
I see, the need for one db call to fetch all quotes makes sense. What happened to the idea of passing in an auction like domain model so that we can apply the fee policies inside this component? 

I'm still not very happy that we are attaching fee policies on the critical path (after the auction deadline has started) where all other "slow" prep work is happening in a background task. I understand it's not the easiest place to add logic, but I feel we are creating significant technical debt here.
Can we not create a different auction struct that is private to the autopilot domain and that allows adding to it before being returned from the SolvableOrdersCache?

We are loading all orders (without caching) from the db, so I think we can also do that with all quotes. This way we wouldn't need a complex `Arc<RWLock<>>` cache inside this component but can instead make its only focus be applying fee policies to auctions/orders. If certain DB queries (e.g quotes for order IDs) are cacheable I think we should have this logic in separate components.